src/meta-srv/src/service/store/ext.rs

```diff
@@ -36,20 +39,19 @@ where
 
         let mut kvs = self.range(req).await?.kvs;
 
-        if kvs.is_empty() {
-            return Ok(None);
-        }
+        Ok(kvs.pop())
+    }
 
-        ensure!(
-            kvs.len() == 1,
-            error::InvalidKvsLengthSnafu {
-                expected: 1_usize,
-                actual: kvs.len(),
-            }
-        );
+    async fn exists(&self, key: Vec<u8>) -> Result<bool> {
+        let req = RangeRequest {
+            key,
+            keys_only: true,
+            ..Default::default()
+        };
+
+        let kvs = self.range(req).await?.kvs;
```

Comment:
So this is not a precise query? E.g., for the following test case `test_exists`, result of "test_key" is `true`.

Reply:
For the `range` API, an empty `range_end` parameter indicates a precise query. However, the unit test did not describe this. To address this issue, I added the "test_key" test which resulted in a false.